gemrb/core/Interface.cpp

```diff
@@ -1849,6 +1850,16 @@ int Interface::Init(InterfaceConfig* config)
 	};
 	EventMgr::RegisterHotKeyCallback(ToggleConsole, ' ', GEM_MOD_CTRL);
 
+	EventMgr::EventCallback appEventCallback = [this](const Event& e) {
+		if (e.type == Event::EventType::RedrawRequest) {
+			winmgr->MarkAllDirty();
+		}
+
+		return true;
+	};
+
+	appEventMonitor = EventMgr::RegisterEventMonitor(appEventCallback, Event::EventTypeMask::AllApplicationMask);
+
```

Comment:
you dont need to do this. Just handle the event in `WindowManager::DispatchEvent`